internal/auth/token.go

```diff
@@ -19,14 +18,9 @@ type TokenResponse struct {
 	ExpiresIn   int    `json:"expires_in"`
 }
 
-type TokenRetriever struct {
-	Authenticator *Authenticator
-	Client        *http.Client
-}
-
-// Refresh gets a new access token from the provided refresh token,
+// RefreshAccessToken gets a new access token from the provided refresh token,
 // The request is used the default client_id and endpoint for device authentication.
-func (t *TokenRetriever) Refresh(ctx context.Context, tenant string) (TokenResponse, error) {
+func RefreshAccessToken(httpClient *http.Client, tenant string) (TokenResponse, error) {
```

Comment:
Wdyt about keeping passing the context over here, instead of the http client and we can use the default http client inside this func until we refactor the code base to have one that is used everywhere across different commands?

Reply:
Removed context because it wasn't being used in this function. HTTP client being passed-in here to enable request mocking for the unit tests. Those mocks had previously been done through the now-deleted `TokenRetriever` abstraction.